src/test/mochitest/browser_dbg-quick-open.js

```diff
@@ -52,64 +57,61 @@ function quickOpen(dbg, query, shortcut = "quickOpen") {
 add_task(async function() {
   const dbg = await initDebugger("doc-script-switching.html");
 
-  info('test opening and closing');
+  info("test opening and closing");
   quickOpen(dbg, "");
   pressKey(dbg, "Escape");
   assertDisabled(dbg);
 
   quickOpen(dbg, "sw");
   pressKey(dbg, "Enter");
 
-  await waitForDispatch(dbg, "LOAD_SOURCE_TEXT");
   let source = dbg.selectors.getSelectedSource(dbg.getState());
   ok(source.get("url").match(/switching-01/), "first source is selected");
 
-  info('Arrow keys and check to see if source is selected');
+  info("Arrow keys and check to see if source is selected");
   quickOpen(dbg, "sw");
   is(resultCount(dbg), 2);
   pressKey(dbg, "Down");
   pressKey(dbg, "Enter");
+  await waitForDispatch(dbg, "SELECT_SOURCE");
```

Comment:
maybe we can make `waitForSelectedSource` smarter... that will help make the tests easier to write

```diff
diff --git a/src/test/mochitest/head.js b/src/test/mochitest/head.js
index 51d659f..e57ff70 100644
--- a/src/test/mochitest/head.js
+++ b/src/test/mochitest/head.js
@@ -218,16 +218,26 @@ async function waitForElement(dbg, selector) {
   return findElementWithSelector(dbg, selector);
 }
 
-function waitForSelectedSource(dbg, sourceId) {
+function waitForSelectedSource(dbg, url) {
   return waitForState(dbg, state => {
     const source = dbg.selectors.getSelectedSource(state);
-    const isLoaded =
-      source && source.has("loadedState") && sourceUtils.isLoaded(source);
-    if (sourceId) {
-      return isLoaded && sourceId == source.get("id");
+    const isLoaded = source && sourceUtils.isLoaded(source);
+    if (!isLoaded) {
+      return false;
     }
 
-    return isLoaded;
+    if (!url) {
+      return true;
+    }
+
+    const newSource = findSource(dbg, url)
+    if (newSource.id != source.get("id")) {
+      return false;
+    }
+
+    const symbols = dbg.selectors.getSymbolsForSource(state, source.id)
+    return symbols;
+
   }, "selected source");
 }
 
```